csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -758,31 +764,42 @@ public sealed class Settings
             /// <summary>
             /// Default settings, as used by <see cref="JsonFormatter.Default"/>
             /// </summary>
-            public static Settings Default { get; }
+            public static Settings Default { get; } = new Settings();
 
-            // Workaround for the Mono compiler complaining about XML comments not being on
-            // valid language elements.
-            static Settings()
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object.
+            /// </summary>
+            public Settings() : this(false, TypeRegistry.Empty)
             {
-                Default = new Settings(false);
             }
 
             /// <summary>
             /// Whether fields whose values are the default for the field type (e.g. 0 for integers)
             /// should be formatted (true) or omitted (false).
             /// </summary>
-            public bool FormatDefaultValues { get; }
+            public bool FormatDefaultValues { get; set; }
 
+            private TypeRegistry typeRegistry_;
             /// <summary>
             /// The type registry used to format <see cref="Any"/> messages.
             /// </summary>
-            public TypeRegistry TypeRegistry { get; }
+            public TypeRegistry TypeRegistry {
```

Comment:
Brace formatting. (The "get" is fine as it is.) Although I'm hoping all of this goes away by it becoming immutable again :)